importer/src/path_canonicalizer.rs

```diff
@@ -312,6 +312,13 @@ fn canonicalize_inside_expression(
                     canonicalize_inside_pattern(pattern, path, paths);
                 })
             }
+            Expression::StructExpression(_, StructExpression { name, .. }) => {
+                if let Some(n) = paths.get(&path.clone().join(name.path.clone())) {
+                    name.path = n.relative_to(&Default::default());
+                } else {
+                    assert!(name.path.try_to_identifier().is_some());
```

Comment:
You copied the code from the Expression::Reference case, right?

This is almost correct, the difference is that in this case, it cannot be a local variable - that would be something like `let x = 8; let y = x{a: 9}`. So you can just use `unwrap()` in lien 316.

Reply:
or `assert!(false)` here...